specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-05-01/sparkConfiguration.json

```diff
@@ -0,0 +1,227 @@
+      "get": {
+        "operationId": "SparkConfigurations_ListByWorkspace",
+        "summary": "List the sparkConfigurations in a workspace.",
+        "description": "List sparkConfigurations in a workspace.",
+        "tags": [
+          "SparkConfigurations"
+        ],
+        "deprecated": false,
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./workspace.json#/parameters/workspaceNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/SparkConfigurationListResponse"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Azure Synapse service.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "consumes": [],
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-examples": {
+          "List sparkConfigurations in a workspace": {
+            "$ref": "./examples/SparkConfigurations_ListByWorkspace.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "SparkConfigurationNameParameter": {
+      "name": "sparkConfigurationName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "SparkConfiguration name",
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "SparkConfigurationListResponse": {
+      "description": "A list of SparkConfiguration resources.",
+      "type": "object",
+      "required": [
+        "value"
+      ],
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of SparkConfiguration.",
+          "items": {
+            "$ref": "#/definitions/SparkConfigurationResource"
+          }
+        },
+        "nextLink": {
+          "description": "The link to the next page of results, if any remaining results exist.",
+          "type": "string"
+        }
+      }
+    },
+    "SparkConfigurationResource": {
+      "type": "object",
+      "description": "SparkConfiguration response details",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
```

Comment:
>"$ref": "#/definitions/SubResource" [](http://example.com/codeflow?start=10&length=35)

I see most synapse resources are derived types of tracked or proxy resources. For example, workspace definition here: [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json#L1128](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json#L1128). 

Make sure SubResource here  is correct

Reply:
Yes, we use this SubResource which we use mostly for DataFactory Resources

Reply:
I see SubResource is only used in [IntegrationRuntimeResource](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/integrationRuntime.json#L2853) . If this is really intended, I would also suggest you to reference the SubResource schema definition in the integrationRuntime.json rather than copying it here.

Reply:
Updated as suggested